vendors/microchip/boards/ecc608a_plus_winsim/aws_demos/application_code/aws_demo_logging.c

```diff
@@ -577,31 +587,37 @@ static void prvLoggingPrintf( uint8_t usLoggingLevel,
         {
             configASSERT( xLogStreamBuffer );
 
-            /* How much space is in the buffer? */
-            xLength2 = uxStreamBufferGetSpace( xLogStreamBuffer );
-
-            /* There must be enough space to write both the string and the length of
-             * the string. */
-            if( xLength2 >= ( xLength + sizeof( xLength ) ) )
+            /* First write in the length of the data, then write in the data
+             * itself.  Raising the thread priority is used as a critical section
+             * as there are potentially multiple writers.  The stream buffer is
+             * only thread safe when there is a single writer (likewise for
+             * reading from the buffer). */
+            xCurrentTask = GetCurrentThread();
+            iOriginalPriority = GetThreadPriority( xCurrentTask );
+            SetThreadPriority( GetCurrentThread(), THREAD_PRIORITY_TIME_CRITICAL );
```

Comment:
As you have said, why not use a FreeRTOS mutex here? 
As far as I know, all the tasks accessing this section will be FreeRTOS tasks which can be handled safely by a mutex. My concern is that raising the priority might not be enough to remove race conditions altogether.

Reply:
A FreeRTOS mutex can only be used by a FreeRTOS task. If a Windows thread calls an API for a FreeRTOS mutex, the kernel will assume that the task identified by `pxCurrentTCB` is waiting for the mutex.
When a FreeRTOS task gets blocked in any of the Windows API's, it may happen that a second FreeRTOS task becomes the active one, causing a big chaos.
The safest option seems to me to play with the current Windows thread priority.